packages/consumption/src/modules/requests/incoming/IncomingRequestsController.ts

```diff
@@ -245,6 +272,150 @@ export class IncomingRequestsController extends ConsumptionBaseController {
         }
     }
 
+    private validateKeyUniquenessOfRelationshipAttributesWithinRequest(
+        items: (RequestItem | RequestItemGroup)[],
+        params: (DecideRequestItemParametersJSON | DecideRequestItemGroupParametersJSON)[]
+    ) {
+        const fragmentsOfMustBeAcceptedItemsOfRequest: RelationshipAttributeFragment[] = [];
+        const fragmentsOfAcceptedItemsOfRequest: RelationshipAttributeFragment[] = [];
+
+        for (let i = 0; i < params.length; i++) {
+            const item = items[i];
+            const decideItemParams = params[i];
+
```

Comment:
You could retrieve the relevant existing attributes here so you don't have to do it in each request item processor

Reply:
On a second thought: With my suggestion the validationResult would not name the problematic requestItems, so I think it's fine to keep them in the itemProcessors. Unless the validationResult is changes to also name the individual problematic requestItems (see the comment for the error message in case of a failed validation)